helion/_compiler/inductor_lowering.py

```diff
@@ -676,6 +840,155 @@ def codegen_baddbmm(ctx: GraphInterpreter, node: torch.fx.Node) -> ast.AST:
     )
 
 
+def mean_helper(
+    x: torch._inductor.ir.TensorBox,
+    axis: list[int] | None = None,
+    keepdim: bool = False,
+    dtype: torch.dtype | None = None,
+    x_size: list[int] | None = None,
+) -> torch._inductor.ir.TensorBox:
+    if dtype is not None:
+        x = to_dtype(x, dtype)
+    if x_size is None:
+        x_size = x.get_size()
+    size = x_size
+    axis = _validate_reduction_axis(x, axis)
+    # compute in higher-precision until end of mean lowering
+    output_dtype = x.get_dtype()
+    if output_dtype in (torch.float16, torch.bfloat16):
+        x = to_dtype(x, torch.float)
+    sum_result = torch._inductor.lowering.sum_(x, axis, keepdim)
+    denom = sympy_product(size[i] for i in axis)
+    denom = ir.IndexingConstant(index=denom, dtype=x.get_dtype(), device=x.get_device())
+    denom = ExpandView.create(denom, list(sum_result.get_size()))
+    return to_dtype(div(sum_result, denom), output_dtype)
+
+
+def var_helper(
+    x: torch._inductor.ir.TensorBox,
+    x_mean: torch._inductor.ir.TensorBox,
+    axis: list[int],
+    keepdim: bool,
+    correction: float | None = None,
+    x_size: list[int] | None = None,
+) -> torch._inductor.ir.TensorBox:
+    if x_size is None:
+        x_size = x.get_size()
+    size = x_size
+
+    diffs = square(sub(x, x_mean))
+    sum_result = sum_(diffs, axis, keepdim)
+
+    denom = sympy_product(size[i] for i in axis)  # pyre-ignore[6]
+    if correction:
+        denom = sympy.Max(denom - correction, 0)
+    denom = ir.IndexingConstant(
+        index=denom,
+        dtype=x.get_dtype(),
+        device=x.get_device(),  # pyre-ignore[6]
+    )
+    denom = ExpandView.create(denom, list(sum_result.get_size()))
+    return div(sum_result, denom)
+
+
+# TODO(yf225): Switch to use upstream torch._inductor.lowering.var_mean_sum_() after PyTorch 2.8 is released.
```

Comment:
If the current direction is a good idea to pursue, I'll open a PR to allow passing in `x_size` in upstream Inductor's `var_mean_sum_()`, and we can switch to use upstream after 2.8 release.